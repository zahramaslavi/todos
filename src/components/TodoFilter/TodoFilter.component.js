import React, { Component } from 'react';

import './TodoFilter.component.scss';

const TodoFilter = ({todoCount, hideCompleted, toggleHideCompleted}) => {

    return (
        <div className="todoFilter">
            <form>
                { 'Todo List(' + todoCount + ')' }
                <input 
                    type="checkbox"  
                    checked={hideCompleted}
                    onChange={() => toggleHideCompleted()}
                />
            </form>
        </div>
    );
}


export default TodoFilter;