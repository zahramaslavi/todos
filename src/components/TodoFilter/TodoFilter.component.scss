.todoFilter {

    form {
        display: flex;
        flex-direction: row;
        justify-content: space-between;
        padding: 1em;
        margin-top: 1em;
        margin-bottom: 1em;
        text-align: left;
        background-color: aliceblue;
        font-size: 1em;
    }
}