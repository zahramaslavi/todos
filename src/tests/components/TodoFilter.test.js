import React from 'react';
import { shallow } from 'enzyme';
import TodoFilter from '../../components/TodoFilter/TodoFilter.component';

test('should test TodoFilter component snapshot', () => {
    const wrapper = shallow(<TodoFilter/>);
    expect(wrapper).toMatchSnapshot();
});

test('should test TodoFilter component snapshot with todoCount', () => {
    const wrapper = shallow(<TodoFilter todoCount={4}/>);
    expect(wrapper).toMatchSnapshot();
});

//todoCount, hideCompleted, toggleHideCompleted
test('should test TodoFilter component snapshot with todoCount and hideCompleted', () => {
    const wrapper = shallow(<TodoFilter todoCount={4} hideCompleted={true}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should test TodoFilter component with toggleHideCompleted event', () => {
    const onSubmitSpy = jest.fn();
    const wrapper = shallow(
        <TodoFilter 
            todoCount={4} 
            hideCompleted={true}
            toggleHideCompleted={onSubmitSpy}
        />);

    const toggleCheckbox = wrapper.find('input[type="checkbox"]');
    toggleCheckbox.simulate('change');
    expect(onSubmitSpy).toHaveBeenCalled();
        
});