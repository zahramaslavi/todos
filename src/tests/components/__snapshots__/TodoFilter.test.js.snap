// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test TodoFilter component snapshot 1`] = `
<div
  className="todoFilter"
>
  <form>
    Todo List(undefined)
    <input
      onChange={[Function]}
      type="checkbox"
    />
  </form>
</div>
`;

exports[`should test TodoFilter component snapshot with todoCount 1`] = `
<div
  className="todoFilter"
>
  <form>
    Todo List(4)
    <input
      onChange={[Function]}
      type="checkbox"
    />
  </form>
</div>
`;

exports[`should test TodoFilter component snapshot with todoCount and hideCompleted 1`] = `
<div
  className="todoFilter"
>
  <form>
    Todo List(4)
    <input
      checked={true}
      onChange={[Function]}
      type="checkbox"
    />
  </form>
</div>
`;
